esp-hal/src/lcd_cam/mod.rs

```diff
@@ -60,7 +111,99 @@ pub enum ByteOrder {
     Inverted = 1,
 }
 
+#[doc(hidden)]
+#[cfg(feature = "async")]
+pub mod asynch {
+    use core::task::Poll;
+
+    use embassy_sync::waitqueue::AtomicWaker;
+    use procmacros::handler;
+
+    use super::private::Instance;
+
+    static TX_WAKER: AtomicWaker = AtomicWaker::new();
+
+    pub(crate) struct LcdDoneFuture {}
+
+    impl LcdDoneFuture {
+        pub(crate) fn new() -> Self {
+            Instance::listen_lcd_done();
```

Comment:
This can be moved to the else branch of the poll method. (I changed the semantics ever so slightly again after you made this future type)